lib/templateFactory.js

```diff
@@ -6,6 +6,9 @@ const compareVersions = require('compare-versions');
 const schema = require('screwdriver-data-schema');
 let instance;
 
+const TEMPLATE_NAME_REGEX = schema.config.regex.FULL_TEMPLATE_NAME;
+const TEMPLATE_NAME_MATCH = 1;
```

Comment:
@FenrirUnbound Writing it that way would mean I'd have a variable defined that's never used (`fullText`) and a couple extra lines to declare the `templateName` and `versionOrTag` variables.

@ian-fox I think it's a good idea to move the magic numbers to the schema. I'll refactor this code after the template tags PRs are merged.